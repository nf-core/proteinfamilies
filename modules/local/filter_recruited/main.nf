process FILTER_RECRUITED {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/83/8372f6241b480332d91bc00a88ec8c72c8f7fcc9994177a5dd67a07007cd6e32/data' :
        'community.wave.seqera.io/library/biopython:1.85--6f761292fa9881b4' }"

    input:
    tuple val(meta), path(domtbl), path(fa)
    val(length_threshold)

    output:
    tuple val(meta), path("${prefix}.fasta.gz"), emit: fasta, optional: true
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    filter_recruited.py \\
        --domtbl ${domtbl} \\
        --fasta ${fa} \\
        --length_threshold ${length_threshold} \\
        --out_fasta ${prefix}.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """
}
