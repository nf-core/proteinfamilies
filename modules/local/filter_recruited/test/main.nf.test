nextflow_process {

    name "Test Process FILTER_RECRUITED"
    script "../main.nf"
    process "FILTER_RECRUITED"

    test("proteinfamilies - hmmsearch - fasta") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'filtering' ],  // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/chunk.domtbl.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = 0.8
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("proteinfamilies - hmmsearch - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'filtering' ],  // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/chunk.domtbl.gz', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = 0.8
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
