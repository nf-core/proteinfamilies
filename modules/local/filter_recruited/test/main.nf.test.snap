{
    "proteinfamilies - hmmsearch - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "filtering"
                        },
                        "filtering_null.fasta.gz:md5,ccf5eb969a8751fe21a8fec277f30953"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b9ebaa78156a9cdd16c4ea79dae93950"
                ],
                "fasta": [
                    [
                        {
                            "id": "filtering"
                        },
                        "filtering_null.fasta.gz:md5,ccf5eb969a8751fe21a8fec277f30953"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b9ebaa78156a9cdd16c4ea79dae93950"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T12:49:18.246171832"
    },
    "proteinfamilies - hmmsearch - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "filtering"
                        },
                        "filtering_null.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b9ebaa78156a9cdd16c4ea79dae93950"
                ],
                "fasta": [
                    [
                        {
                            "id": "filtering"
                        },
                        "filtering_null.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b9ebaa78156a9cdd16c4ea79dae93950"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T12:49:39.946682352"
    }
}